highlighters/main/main-highlighter.zsh

```diff
@@ -795,18 +789,72 @@ _zsh_highlight_main_highlighter_check_path()
   return 1
 }
 
+# Highlight an argument and possibly special chars in quotes
+_zsh_highlight_main_highlighter_highlight_argument()
+{
+  local i
+
+  _zsh_highlight_main_add_region_highlight $start_pos $end_pos default
+  for (( i = 1 ; i <= end_pos - start_pos ; i += 1 )); do
+    case "$arg[$i]" in
+      "\\") (( i += 1 )); continue;;
+      "'") _zsh_highlight_main_highlighter_highlight_single_quote $i; (( i = REPLY ));;
+      '"') _zsh_highlight_main_highlighter_highlight_double_quote $i; (( i = REPLY ));;
+      '$')
+        if [[ $arg[i+1] == "'" ]]; then
+          _zsh_highlight_main_highlighter_highlight_dollar_quote $i
+          (( i = REPLY ))
+        elif [[ $arg[i+1] == [\^=~#+] ]]; then
+          while [[ $arg[i+1] == [\^=~#+] ]]; do
+            (( i += 1 ))
+          done
+          if [[ $arg[i+1] == [*@#?-$!] ]]; then
+            (( i += 1 ))
+          fi
+        elif [[ $arg[i+1] == [*@#?-$!] ]]; then
+          (( i += 1 ))
+        fi;;
+      [*?])
+        if $highlight_glob; then
+          _zsh_highlight_main_add_region_highlight $start_pos $end_pos globbing
+          break
+        fi;;
+      *) continue;;
```

Comment:
A potential enhancement (for a separate PR :)) would be to also find `{brace,expansions}` and `[character_classes]` here.  (And maybe more exotic forms of globbing, such as `foo(bar|baz)qux`, as well.)